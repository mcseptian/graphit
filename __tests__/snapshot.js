// __tests__/snapshot.js

import React from "react";
import { create } from "react-test-renderer";
import Home from "../pages/index";
import { MockedProvider } from "@apollo/client/testing";
import { InMemoryCache, gql } from "@apollo/client";

const cache = new InMemoryCache();
cache.writeQuery({
  query: gql`
    query pokemons($limit: Int, $offset: Int) {
      pokemons(limit: $limit, offset: $offset) {
        count
        next
        previous
        status
        message
        results {
          url
          name
          image
        }
      }
    }
  `,

  data: {
    pokemons: {
      count: 1118,
      next: "https://pokeapi.co/api/v2/pokemon/?offset=3&limit=2",
      previous: "https://pokeapi.co/api/v2/pokemon/?offset=0&limit=1",
      status: true,
      message: "",
      results: [
        {
          url: "https://pokeapi.co/api/v2/pokemon/2/",
          name: "ivysaur",
          image:
            "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png",
        },
        {
          url: "https://pokeapi.co/api/v2/pokemon/3/",
          name: "venusaur",
          image:
            "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/3.png",
        },
      ],
    },
  },
});

const useRouter = jest.spyOn(require("next/router"), "useRouter");

it("renders homepage unchanged", () => {
  useRouter.mockImplementationOnce(() => ({
    pathname: "/my-list",
  }));

  const tree = create(
    <MockedProvider cache={cache}>
      <Home />
    </MockedProvider>
  );

  expect(tree.toJSON()).toMatchSnapshot();
});
