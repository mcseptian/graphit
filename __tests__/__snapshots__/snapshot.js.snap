// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
Array [
  <header
    css={
      Object {
        "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkhlYWRlci5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFTYyIsImZpbGUiOiJIZWFkZXIuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgeyB1c2VSb3V0ZXIgfSBmcm9tIFwibmV4dC9yb3V0ZXJcIjtcbmltcG9ydCBMaW5rIGZyb20gXCJuZXh0L2xpbmtcIjtcbmltcG9ydCB7IGNzcywganN4IH0gZnJvbSBcIkBlbW90aW9uL3JlYWN0XCI7XG5cbmV4cG9ydCBkZWZhdWx0IGZ1bmN0aW9uIEhlYWRlcigpIHtcbiAgY29uc3QgeyBwYXRobmFtZSB9ID0gdXNlUm91dGVyKCk7XG5cbiAgcmV0dXJuIChcbiAgICA8aGVhZGVyXG4gICAgICBjc3M9e2Nzc2BcbiAgICAgICAgaGVhZGVyIHtcbiAgICAgICAgICBtYXJnaW4tYm90dG9tOiAyNXB4O1xuICAgICAgICB9XG4gICAgICAgIGEge1xuICAgICAgICAgIGZvbnQtc2l6ZTogMTRweDtcbiAgICAgICAgICBtYXJnaW4tcmlnaHQ6IDE1cHg7XG4gICAgICAgICAgdGV4dC1kZWNvcmF0aW9uOiBub25lO1xuICAgICAgICB9XG4gICAgICAgIC5pcy1hY3RpdmUge1xuICAgICAgICAgIHRleHQtZGVjb3JhdGlvbjogdW5kZXJsaW5lO1xuICAgICAgICB9XG4gICAgICBgfVxuICAgID5cbiAgICAgIDxMaW5rIGhyZWY9XCIvXCI+XG4gICAgICAgIDxhIGNsYXNzTmFtZT17cGF0aG5hbWUgPT09IFwiL1wiID8gXCJpcy1hY3RpdmVcIiA6IFwiXCJ9PkhvbWU8L2E+XG4gICAgICA8L0xpbms+XG4gICAgICA8TGluayBocmVmPVwiL215LWxpc3RcIj5cbiAgICAgICAgPGEgY2xhc3NOYW1lPXtwYXRobmFtZSA9PT0gXCIvbXktbGlzdFwiID8gXCJpcy1hY3RpdmVcIiA6IFwiXCJ9Pk15LUxpc3Q8L2E+XG4gICAgICA8L0xpbms+XG4gICAgPC9oZWFkZXI+XG4gICk7XG59XG4iXX0= */",
        "name": "4cajdb-Header",
        "styles": "header{margin-bottom:25px;}a{font-size:14px;margin-right:15px;text-decoration:none;}.is-active{text-decoration:underline;};label:Header;",
        "toString": [Function],
      }
    }
  >
    <a
      className=""
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Home
    </a>
    <a
      className="is-active"
      href="/my-list"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      My-List
    </a>
  </header>,
  <div
    css={
      Object {
        "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkluZm9Cb3guanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBWWMiLCJmaWxlIjoiSW5mb0JveC5qcyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCB7IGNzcywganN4IH0gZnJvbSBcIkBlbW90aW9uL3JlYWN0XCI7XG5pbXBvcnQgeyB1c2VTdGF0ZSB9IGZyb20gXCJyZWFjdFwiO1xuXG5jb25zdCBJbmZvQm94ID0gKHsgY2hpbGRyZW4gfSkgPT4ge1xuICBjb25zdCBbaW5mb0Rpc3BsYXksIHNldEluZm9EaXNwbGF5XSA9IHVzZVN0YXRlKHRydWUpO1xuXG4gIGNvbnN0IGhhbmRsZUNsaWNrID0gKCkgPT4ge1xuICAgIHNldEluZm9EaXNwbGF5KGZhbHNlKTtcbiAgfTtcblxuICByZXR1cm4gKFxuICAgIDxkaXZcbiAgICAgIGNzcz17Y3NzYFxuICAgICAgICBkaXNwbGF5OiAke2luZm9EaXNwbGF5ID8gXCJibG9ja1wiIDogXCJub25lXCJ9O1xuICAgICAgICBtYXJnaW4tdG9wOiAyMHB4O1xuICAgICAgICBtYXJnaW4tYm90dG9tOiAyMHB4O1xuICAgICAgICBwYWRkaW5nLXRvcDogMjBweDtcbiAgICAgICAgcGFkZGluZy1ib3R0b206IDIwcHg7XG4gICAgICAgIGJvcmRlci10b3A6IDFweCBzb2xpZCAjZWNlY2VjO1xuICAgICAgICBib3JkZXItYm90dG9tOiAxcHggc29saWQgI2VjZWNlYztcbiAgICAgIGB9XG4gICAgPlxuICAgICAge2NoaWxkcmVufVxuICAgICAgPGJ1dHRvblxuICAgICAgICBjc3M9e2Nzc2BcbiAgICAgICAgICBwYWRkaW5nOiAxMHB4IDIwcHg7XG4gICAgICAgICAgYm9yZGVyLXJhZGl1czogNXB4XG4gICAgICAgICAgYm9yZGVyLWNvbG9yOiAjZmZmO1xuICAgICAgICAgIG91dGxpbmU6IG5vbmU7XG4gICAgICAgIGB9XG4gICAgICAgIG9uQ2xpY2s9e2hhbmRsZUNsaWNrfVxuICAgICAgPlxuICAgICAgICBDbG9zZVxuICAgICAgPC9idXR0b24+XG4gICAgPC9kaXY+XG4gICk7XG59O1xuXG5leHBvcnQgZGVmYXVsdCBJbmZvQm94O1xuIl19 */",
        "name": "k37fz5-InfoBox",
        "next": undefined,
        "styles": "display:block;margin-top:20px;margin-bottom:20px;padding-top:20px;padding-bottom:20px;border-top:1px solid #ececec;border-bottom:1px solid #ececec;;label:InfoBox;",
        "toString": [Function],
      }
    }
  >
    This page show a list of Pokemons' names and the owned total. Click on the list item to see Pokemon Detail.
    <button
      css={
        Object {
          "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkluZm9Cb3guanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBd0JnQiIsImZpbGUiOiJJbmZvQm94LmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IHsgY3NzLCBqc3ggfSBmcm9tIFwiQGVtb3Rpb24vcmVhY3RcIjtcbmltcG9ydCB7IHVzZVN0YXRlIH0gZnJvbSBcInJlYWN0XCI7XG5cbmNvbnN0IEluZm9Cb3ggPSAoeyBjaGlsZHJlbiB9KSA9PiB7XG4gIGNvbnN0IFtpbmZvRGlzcGxheSwgc2V0SW5mb0Rpc3BsYXldID0gdXNlU3RhdGUodHJ1ZSk7XG5cbiAgY29uc3QgaGFuZGxlQ2xpY2sgPSAoKSA9PiB7XG4gICAgc2V0SW5mb0Rpc3BsYXkoZmFsc2UpO1xuICB9O1xuXG4gIHJldHVybiAoXG4gICAgPGRpdlxuICAgICAgY3NzPXtjc3NgXG4gICAgICAgIGRpc3BsYXk6ICR7aW5mb0Rpc3BsYXkgPyBcImJsb2NrXCIgOiBcIm5vbmVcIn07XG4gICAgICAgIG1hcmdpbi10b3A6IDIwcHg7XG4gICAgICAgIG1hcmdpbi1ib3R0b206IDIwcHg7XG4gICAgICAgIHBhZGRpbmctdG9wOiAyMHB4O1xuICAgICAgICBwYWRkaW5nLWJvdHRvbTogMjBweDtcbiAgICAgICAgYm9yZGVyLXRvcDogMXB4IHNvbGlkICNlY2VjZWM7XG4gICAgICAgIGJvcmRlci1ib3R0b206IDFweCBzb2xpZCAjZWNlY2VjO1xuICAgICAgYH1cbiAgICA+XG4gICAgICB7Y2hpbGRyZW59XG4gICAgICA8YnV0dG9uXG4gICAgICAgIGNzcz17Y3NzYFxuICAgICAgICAgIHBhZGRpbmc6IDEwcHggMjBweDtcbiAgICAgICAgICBib3JkZXItcmFkaXVzOiA1cHhcbiAgICAgICAgICBib3JkZXItY29sb3I6ICNmZmY7XG4gICAgICAgICAgb3V0bGluZTogbm9uZTtcbiAgICAgICAgYH1cbiAgICAgICAgb25DbGljaz17aGFuZGxlQ2xpY2t9XG4gICAgICA+XG4gICAgICAgIENsb3NlXG4gICAgICA8L2J1dHRvbj5cbiAgICA8L2Rpdj5cbiAgKTtcbn07XG5cbmV4cG9ydCBkZWZhdWx0IEluZm9Cb3g7XG4iXX0= */",
          "name": "1ltt28h-InfoBox",
          "styles": "padding:10px 20px;border-radius:5px border-color:#fff;outline:none;label:InfoBox;",
          "toString": [Function],
        }
      }
      onClick={[Function]}
    >
      Close
    </button>
  </div>,
  <div>
    Loading
  </div>,
]
`;
